app/components/ChainsDropdown/ChainsDropdown.tsx

```diff
@@ -25,59 +42,88 @@ SelectItem.displayName = "SelectItem"
 
 const ChainsDropdown = ({
   chains,
+  checkboxData,
+  endpoint,
   onChange,
 }: {
   chains: BlockchainsQuery["blockchains"]
+  checkboxData?: string[]
+  endpoint: EndpointQuery["endpoint"]
   onChange: (chain: string) => void
 }) => {
+  const [isInputShown, setIsInputShown] = useState(false)
   const theme = useMantineTheme()
+  const selectRef = useRef<HTMLDivElement>(null)
   const addNewChainSelectRef = useRef<HTMLInputElement>(null)
+  const user = useUser()
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (selectRef.current && !selectRef.current.contains(event.target as Node)) {
+        setIsInputShown(false)
+      }
+    }
+    document.addEventListener("mousedown", handleClickOutside)
+
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside)
+    }
+  }, [])
 
   const selectChainData = useMemo(() => {
     return chains
       .map((chain) => ({
         label: chain?.description ?? "",
         value: chain?.id ?? "",
+        checked:
+          (checkboxData &&
+            checkboxData?.length &&
+            checkboxData.includes(chain?.id ?? "")) ??
+          false,
       }))
-      .sort((a, b) => {
-        if (a.label < b.label) {
-          return -1
-        }
-        if (a.label > b.label) {
-          return 1
-        }
-        return 0
-      })
-  }, [chains])
+      .sort((a, b) => a.label.localeCompare(b.label))
+  }, [chains, user, endpoint])
 
-  return (
-    <Select
-      ref={addNewChainSelectRef}
-      searchable
-      aria-label="Add new"
-      data={selectChainData}
-      icon={<IconSearch fill={theme.colors.blue[5]} height={18} width={18} />}
-      itemComponent={SelectItem}
-      placeholder="Add new"
-      rightSection={<></>}
-      rightSectionWidth={0}
+  return isInputShown || !checkboxData ? (
+    <div ref={selectRef}>
+      <Select
+        ref={addNewChainSelectRef}
+        searchable
+        allowDeselect={true}
+        aria-label="Search Network"
+        data={selectChainData}
+        icon={<IconSearch fill={theme.colors.blue[5]} height={18} width={18} />}
+        itemComponent={checkboxData ? CheckboxItem : SelectItem}
+        placeholder="Search Network"
+        rightSectionWidth={0}
+        size="xs"
+        sx={(theme: MantineTheme) => ({
```

Comment:
Is this checkbox styling going to be applied to all checkboxes in the app? if yes I suggest to move this to a standalone component and return the styled component or define the styles in a shared place.